src/components/v2/V2Project/V2FundingCycleSection/modals/EditTokenAllocationModal.tsx

```diff
@@ -0,0 +1,112 @@
+import { t, Trans } from '@lingui/macro'
+import { Form, Modal } from 'antd'
+import { Split } from 'models/v2/splits'
+import { useCallback, useContext, useEffect, useMemo, useState } from 'react'
+import { FormItems } from 'components/shared/formItems'
+
+import { toMod, toSplit } from 'utils/v2/splits'
+
+import { V2ProjectContext } from 'contexts/v2/projectContext'
+
+import { useSetProjectSplits } from 'hooks/v2/transactor/SetProjectSplits'
+
+import { preciseFormatSplitPercent } from 'utils/v2/math'
+
+import { RESERVED_TOKEN_SPLIT_GROUP } from 'constants/v2/splits'
+
+export const EditTokenAllocationModal = ({
+  visible,
+  onOk,
+  onCancel,
+}: {
+  visible: boolean
+  onOk: VoidFunction
+  onCancel: VoidFunction
+}) => {
+  const { reservedTokensSplits, fundingCycleMetadata, fundingCycle } =
+    useContext(V2ProjectContext)
+  const reservedRate = fundingCycleMetadata?.reservedRate
+
+  const setProjectSplits = useSetProjectSplits({
+    domain: fundingCycle?.configuration?.toString(),
+  })
+
+  const [editingReservedTokensSplits, setEditingReservedTokensSplits] =
+    useState<Split[]>([])
+  const [modalLoading, setModalLoading] = useState<boolean>(false)
+
+  useEffect(() => {
+    if (!reservedTokensSplits) return
+    setEditingReservedTokensSplits(reservedTokensSplits)
+  }, [reservedTokensSplits])
+
+  const totalPercentage = useMemo(
+    () =>
+      editingReservedTokensSplits
+        ?.map(s => preciseFormatSplitPercent(s.percent))
+        .reduce((a, b) => a + b, 0) ?? 0,
+    [editingReservedTokensSplits],
+  )
+  const totalPercentagesInvalid = totalPercentage > 100
+
+  const onSaveTokenAllocation = useCallback(async () => {
+    if (totalPercentagesInvalid) return
+    setModalLoading(true)
+    const tx = await setProjectSplits(
+      {
+        groupedSplits: {
+          group: RESERVED_TOKEN_SPLIT_GROUP,
+          splits: editingReservedTokensSplits ?? [],
+        },
+      },
+      {
+        onConfirmed: () => {
+          setModalLoading(false)
+          onOk()
+        },
+        onError: () => setModalLoading(false),
+      },
+    )
+    if (!tx) {
+      setModalLoading(false)
+    }
+  }, [
+    editingReservedTokensSplits,
+    onOk,
+    setProjectSplits,
+    totalPercentagesInvalid,
+  ])
+
+  return (
+    <Modal
+      visible={visible}
+      confirmLoading={modalLoading}
+      title={t`Edit reserved token allocation`}
+      okText={t`Save token allocation`}
+      cancelText={modalLoading ? t`Close` : t`Cancel`}
+      width={720}
+      onOk={() => onSaveTokenAllocation()}
+      onCancel={onCancel}
+      okButtonProps={{ disabled: totalPercentagesInvalid }}
+    >
+      <Form layout="vertical">
```

Comment:
**suggestion** WDYT about slapping a `Callout` to the top of the modal explaining that changes take effect immediately.

```
Changes to your reserved token allocation will take effect immediately.
```

Reply:
What is a callout?

Reply:
This guy https://github.com/jbx-protocol/juice-interface/blob/b8143219ff0e8957022b525b8c4439af56f44744/src/components/shared/Callout.tsx

Reply:
<img width="702" alt="image" src="https://user-images.githubusercontent.com/104132113/175893402-8e37a4a5-e398-42d4-b020-73ac13a58493.png">